src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2558,20 +2558,65 @@ exprt c_typecheck_baset::do_special_functions(
     return std::move(malloc_expr);
   }
   else if(
-    identifier == CPROVER_PREFIX "r_ok" || identifier == CPROVER_PREFIX "w_ok")
+    identifier == CPROVER_PREFIX "r_ok" ||
+    identifier == CPROVER_PREFIX "w_ok" || identifier == CPROVER_PREFIX "rw_ok")
   {
-    if(expr.arguments().size() != 2)
+    if(expr.arguments().size() != 1 && expr.arguments().size() != 2)
```

Comment:
Minor, but this is using our old constraint handling infrastructure - would it be better if this was written as two preconditions? Something like this:

```C
PRECONDITION(expr.arguments().size() != 1);
PRECONDITION(expr.arguments().size() != 2);
```

or

```c
INVARIANT(expr.arguments().size() != 1 || expr.arguments().size() != 2, "r_ok/w_ok/rw_ok expect one or two operands");
```

Reply:
No, these aren't preconditions, but checks on user input.